#!/bin/bash
#LloydJonesIII


## user input 
read -p "Please name your Project: " pname
read -p "Please enter the project ID you are researching: " project


## Make Directories 
mkdir ../../media/volume/sdb/$pname
mkdir ../../media/volume/sdb/$pname/fastq
mkdir ../../media/volume/sdb/$pname/assembly
mkdir ../../media/volume/sdb/$pname/fastq/untrimmed
mkdir ../../media/volume/sdb/$pname/fastq/trimmed
mkdir ../../media/volume/sdb/$pname/assembly/results
mkdir ../../media/volume/sdb/$pname/assembly/results/SAM
mkdir ../../media/volume/sdb/$pname/assembly/results/BAM
mkdir ../../media/volume/sdb/$pname/assembly/results/BCF
mkdir ../../media/volume/sdb/$pname/assembly/results/VCF


## NIH API Websearch
esearch -db sra -query $project | efetch -format docsum | xtract -pattern Runs -ACC @acc  -element "&ACC" > ../../media/volume/sdb/$pname/$pname.txt


## Grab metadata using the esearch list 
prefetch --option-file $pname.txt -O ../../media/volume/sdb/$pname


## takes list generated by esearch to grab the desired fastq files from the databases
fasterq-dump --outdir ../../media/volume/sdb/$pname/fastq ../../media/volume/sdb/$pname/SRR*

## gzip all fastq files that were collected 
gzip ../../media/volume/sdb/$pname/fastq/*.fastq



